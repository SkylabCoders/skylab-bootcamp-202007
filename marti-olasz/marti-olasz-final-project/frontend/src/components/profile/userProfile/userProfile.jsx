import React, { useEffect } from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import { withAuthenticationRequired } from '@auth0/auth0-react';
import { connect } from 'react-redux';

import { DOMAIN } from '../../../config/auth0';
import {
  saveUser,
  getUser,
  userEdit
} from '../../../redux/actions/authActions';

import UserHeader from './userHeader/userHeader';
import UserBio from './userBio/userBio';
import Following from './following/following';

import PuffLoader from 'react-spinners/PuffLoader';

import './userProfile.scss';

const UserProfile = ({ mongoUser, dispatch }) => {
  const { user, isAuthenticated, getAccessTokenSilently } = useAuth0();

  useEffect(() => {
    (async () => {
      try {
        const accessToken = await getAccessTokenSilently({
          audience: `https://${DOMAIN}/api/v2/`,
          scope: 'read:current_user'
        });
        if (accessToken) {
          sessionStorage.setItem('token', JSON.stringify(accessToken));
          dispatch(userEdit({}));
          dispatch(saveUser(user.sub));
          dispatch(getUser(user));
        }
      } catch (err) {
        console.log(err.message);
      }
    })();
  }, [dispatch, getAccessTokenSilently, user]);

  return isAuthenticated && mongoUser && user ? (
    <div className='user-profile'>
      <UserHeader />
      <div className='user-profile__bottom'>
        <UserBio
          bio={mongoUser.bio}
          name={mongoUser.user}
          dispatch={dispatch}
        />
        <Following following={mongoUser.following} />
      </div>
    </div>
  ) : (
    <PuffLoader color='#f55110' />
  );
};

function mapStateToProps(state) {
  return {
    mongoUser: state.authReducer.user
  };
}

export default connect(mapStateToProps)(
  withAuthenticationRequired(UserProfile, {
    onRedirecting: () => <PuffLoader color='#f55110' />
  })
);
export { UserProfile };
