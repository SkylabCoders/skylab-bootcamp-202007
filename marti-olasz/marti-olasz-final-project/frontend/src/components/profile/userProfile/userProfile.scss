$color-detail: #f55110;

.user-profile {
  min-width: 500px;
  display: flex;
  flex-direction: column;
  color: white;
  padding: 20px 10%;
  padding-bottom: 50px;

  .user-profile__bottom {
    display: flex;
  }
}
