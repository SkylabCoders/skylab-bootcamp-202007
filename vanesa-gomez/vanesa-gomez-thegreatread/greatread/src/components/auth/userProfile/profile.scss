* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Lato', sans-serif;
    text-decoration: none;
}

.profile-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    .image-container {
        display: flex;
        justify-content: center;
        align-items: center;

        img {
            border-radius: 50%;
            margin-top: 20px;
            margin-bottom: 20px;
            width: 200px;
        }
    }

    .nickname-container {
        display: flex;
        justify-content: center;
        margin-top: 25px;

        .nickname-item {
            font-size: 23px;
        }
    }

    .favorite-books-container {
        padding-top: 30px;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        margin-top: 10px;
        background-color: #f5f5f5;

        .pageHeadLine {
            font-size: 12px;
            margin-bottom: 10px;
        }

        .bookShelves {
            display: flex;
            flex-direction: column;
            margin: 10px;

            .bookShelfList {
                display: flex;

                .book {
                    display: flex;

                    .bookCover {
                        padding: 10px;
                    }

                    .book-details {
                        padding: 10px;
                        text-decoration: none;
                        color: black;
                        width: 50%;

                        .bookTitle {
                            font-size: 15px;
                            font-weight: 600;
                        }

                        .bookAuthor {
                            color: grey;
                            font-style: italic;
                            font-size: 13px;
                            margin-top: 2px;
                        }

                        .bookRating {
                            font-size: 30px;
                            color: #e74a10;
                            margin-top: 4px;
                        }
                    }
                }
            }
        }
    }
}

// Media queries
@media screen and (min-width: 1024px) {
    .favorite-books-container {
        width: 90%;
    }

    .bookShelves {
        display: flex;
        flex-direction: row !important;
        width: 100%;
        overflow-x: auto;
    }

    .bookShelves::-webkit-scrollbar {
        display: none;
    }
}
