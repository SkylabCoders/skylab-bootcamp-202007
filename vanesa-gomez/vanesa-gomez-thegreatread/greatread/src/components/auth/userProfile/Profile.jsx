import React, { useState, useEffect } from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import './profile.scss';
import { createUser, loadUser } from '../../../actions/userActions';
import userStore from '../../../stores/userStore';
import LoadingPage from '../../LoadingPage/LoadingPage';
import { Link } from 'react-router-dom';

const Profile = (props) => {
    const { user, isAuthenticated, isLoading } = useAuth0();
    const [userLoaded, setUserLoaded] = useState(userStore.getUser());

    useEffect(() => {
        userStore.addChangeListener(onChange);

        if (user) {
            loadUser(user?.sub);
        }
        return () => userStore.removeChangeListener(onChange);
    }, [user]);

    function onChange() {
        setUserLoaded(userStore.getUser());
    }

    if (isAuthenticated) {
        createUser({
            userEmail: user.email,
            sub: user.sub,
            userNickname: user.nickname
        });
    }
    if (isLoading) {
        return <LoadingPage />;
    }

    return (
        isAuthenticated &&
        userLoaded && (
            <div className="profile-container">
                <div className="nickname-container">
                    <h1 className="nickname-item"> Hola {user.nickname}!</h1>
                </div>
                <div className="image-container">
                    <img src={user.picture} alt="Imagen de perfil de usuario" />
                </div>
                {userLoaded.favoriteBooks.length > 0 && (
                    <div className="favorite-books-container">
                        <div className="pageHeadLine">
                            <h1>Mis libros favoritos</h1>
                        </div>
                        <div className="bookShelves">
                            {userLoaded &&
                                userLoaded.favoriteBooks?.map((book) => (
                                    <div
                                        className="bookShelfList"
                                        key={`${book.id}`}
                                    >
                                        <div
                                            to={`/book/${book.id}`}
                                            className="book"
                                        >
                                            <Link
                                                to={`/book/${book.id}`}
                                                className="bookCover"
                                            >
                                                <img
                                                    src={`${book.image}`}
                                                    alt="Cover libro"
                                                    width="200"
                                                    height="300"
                                                />
                                            </Link>
                                            <div className="book-details">
                                                <p className="bookTitle">
                                                    {book.title}
                                                </p>
                                                <p className="bookAuthor">
                                                    por {book.author}
                                                </p>
                                                <p className="bookRating">
                                                    {book.averageRating}
                                                </p>
                                            </div>
                                        </div>
                                    </div>
                                ))}
                        </div>
                    </div>
                )}
            </div>
        )
    );
};

export default Profile;
