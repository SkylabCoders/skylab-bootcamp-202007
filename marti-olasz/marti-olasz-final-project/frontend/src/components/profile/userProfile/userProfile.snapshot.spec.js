import React from 'react';
import renderer from 'react-test-renderer';

import { UserProfile } from './userProfile';

jest.mock('./following/following');

describe('UserProfile snapshot', () => {
  const mongoUser = {
    bio: 'bio',
    user: 'user',
    following: []
  };
  const dispatch = jest.fn();
  it('Should match', () => {
    const tree = renderer.create(
      <UserProfile mongoUser={mongoUser} dispatch={dispatch} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
