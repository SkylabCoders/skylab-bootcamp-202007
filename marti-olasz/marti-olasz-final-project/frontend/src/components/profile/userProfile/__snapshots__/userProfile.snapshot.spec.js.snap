// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile snapshot Should match 1`] = `
<div
  className="css-3fvcat"
>
  <div
    className="css-hwvl8z"
  />
  <div
    className="css-16hco3u"
  />
</div>
`;
